ClusteringAlgsComparisons/ClusteringComparison.py

```diff
@@ -64,15 +95,24 @@ def __do_cluster(self, matrix: np.ndarray, alg_func) -> List[Set[int]]:
     def __do_regret(matrix: np.ndarray, clusters: List[Set[int]]) -> float:
         return DataLossEstimator.calculate_data_loss(matrix, clusters)
 
-    def compare(self, alg_dict: dict) -> dict:
+    def compare(self, alg_dict: dict, matrices_ids: list = None, matrix_type: str = 'scatter_based'):
         for key in alg_dict.keys():
             self.data[key] = Result(self.tests_num)
 
         for test_iter in np.arange(self.tests_num):
             print("Running test {0}".format(test_iter + 1))
-            block_matrix = self.create_block_matrix() if self.matrix is None else self.matrix
-            reduced_matrix = self.reduce_matrix(block_matrix)
+
+            if matrix_type == 'scatter_based':
+                matrix = self.create_scatter_based_matrix() if self.matrix is None else self.matrix
+            elif matrix_type == 'block':
+                matrix = self.create_block_matrix() if self.matrix is None else self.matrix
+            else:
+                matrix = self.create_block_matrix() if self.matrix is None else self.matrix
+
+            reduced_matrix = self.reduce_matrix(matrix)
```

Comment:
This doesn't cause everything to fail almost by accident - 
Since the missions_num is in effect 1 (see implementation of self.reduce_matrix() and the initializations of ClusteringComparator in ClusteringTest).